atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceTransactionIntegrationTest.java

```diff
@@ -60,6 +73,28 @@ protected KeyValueService getKeyValueService() {
         return kvsSupplier.get();
     }
 
+    @Test
+    public void canUntangleHighlyConflictingPutUnlessExists() {
+        TransactionTables.createTables(keyValueService);
+        TransactionTables.truncateTables(keyValueService);
+        TransactionService transactionService = WriteBatchingTransactionService.create(
+                SimpleTransactionService.createV2(keyValueService));
+        ExecutorService executorService = Executors.newCachedThreadPool();
+        List<Future<?>> futures = LongStream.range(0, 100)
+                .mapToObj(unused -> executorService.submit(() -> {
```

Comment:
unused 🗡

Reply:
I think the 😅 emoji is appropriate here - the original test had random numbers, but I found this pretty reliably reproduced the issue.

Reply:
also changing the numbers somewhat, as `timestamp - 1` means we might end up with negative timestamps.